File: frontend/src/app/services/config/departments.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { ConfigService } from './config.service';
import { Manager } from 'src/app/model/config/manager.model';
import { AppResponse } from 'src/app/dto/response.dto';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class DepartmentsService {


  constructor(private http: HttpClient, private config: ConfigService) { }

  getManagers(): Observable<AppResponse<Manager[]>> {
    const url = `${this.config.apiUrl}/departments/managers`;
    return this.http.get<AppResponse<Manager[]>>(url);
  }
}

File: backend/src/main/java/com/prologiccreations/traderssolution/service/super_classes/config/DepartmentsService.java
package com.prologiccreations.traderssolution.service.super_classes.config;

import com.prologiccreations.traderssolution.dto.Response;
import com.prologiccreations.traderssolution.dto.config.ManagerDto;
import com.prologiccreations.traderssolution.model.config.Department;
import com.prologiccreations.traderssolution.service.super_classes.CrudService;

import java.util.List;

public interface DepartmentsService extends CrudService<Department, Long> {

    Response<List<ManagerDto>> findAllManagers();

}
